### PR TITLE
Add more options to the related articles plugin

### PR DESCRIPTION
**Pull request by Will-Hoey**
Some improvements to the related articles plugin based on https://app.forecast.it/project/P-185/workflow/T12669

### CODE REVIEW

File: news/models.py [hunk 1]
```diff
@@ -177,17 +185,54 @@ def copy_relations(self, oldinstance):
         """
         self.tags.set(oldinstance.tags.all())
 
-    def get_latest_articles(self, limit=3):
+    @property
+    def category_articles(self):
+        """
+        Return a queryset for a category
+        """
+        return (
```
Comment (michaeljcollinsuk) [inline news/models.py:209]: Is it Black that adds the parenthises around this?

Comment (Will-Hoey) [reply news/models.py:209]: Yeah. It's funky about having a queryset like this split over multiple lines

File: news/models.py [hunk 2]
```diff
@@ -177,17 +185,54 @@ def copy_relations(self, oldinstance):
...
+            Article.objects.published()
+            .filter(category__in=self.category)
+            .distinct()
+            .order_by("-created_at")[: self.num_articles]
```
Comment (michaeljcollinsuk) [inline news/models.py]: You could remove `[:self.num_articles]` from here and other methods and instead add it on to the return in your `get_articles` method so you only have it once `queryset[:self.num_articles]`

File: news/models.py [hunk 3]
```diff
@@ -177,17 +185,54 @@ def copy_relations(self, oldinstance):
...
+
+    def get_articles(self):
+        """
+        Return a queryset based on what the user chooses on the frontend
+        """
+        queryset = self.articles_recent
+
+        if self.tags.all():
+            queryset = self.tags_articles
+        if self.category:
```
Comment (michaeljcollinsuk) [inline news/models.py:243]: So this is an either/or e.g if you have a category set, that will override your previous filtering by tags? So you cant have articles filtered by both? Is that the intention?

File: news/models.py (hunk 1 above)
Comment (michaeljcollinsuk) [reply news/models.py:209]: Thought as much

File: news/models.py (hunk 3 above)
Comment (Will-Hoey) [reply news/models.py:243]: It's the intention for now yes. There's a priority in what queryset is returned, although I should add help text so the user knows

Comment (michaeljcollinsuk) [reply news/models.py:243]: Ok yeah - if thats what they want thats fine, but update the helptext to explain how it works. If you did want to filter by both, it wouldnt need too much to update, you would just want to pass your initial `queryset` in to the filter methods

File: news/models.py [hunk 4]
```diff
@@ -177,17 +185,54 @@ def copy_relations(self, oldinstance):
         """
         self.tags.set(oldinstance.tags.all())
 
-    def get_latest_articles(self, limit=3):
+    @property
+    def category_articles(self):
```
Comment (michaeljcollinsuk) [inline news/models.py]: This name could be more descriptive to reflect what it does - it filters articles by category. So `filtered_by_category` perhaps?

**Review by michaeljcollinsuk — APPROVED** [review]

**Pull request merged**